docs/preset-typescript.md

```diff
@@ -58,11 +58,23 @@ require("@babel/core").transform("code", {
 
 ## Options
 
+### `isTSX`
+
+`boolean`, defaults to `false`.
+
+Forcibly enables `jsx` parsing. Otherwise angle brackets will be treated as typescript's legacy type assertion `var foo = <string>bar;`. Also, `isTSX: true` requires `allExtensions: true`
+
 ### `jsxPragma`
 
-`string`
+`string`, defaults to `React`.
 
 Replace the function used when compiling JSX expressions.
 
+### `allExtensions`
+
+`boolean`, defaults to `false`.
+
+If `true`, applies `jsxPragma` and `isTSX` options to both `.ts` and `.tsx` files. Otherwise `isTSX` option will be applied only to `.tsx` files.  
```

Comment:
Actually this option indicates that every file (`.ts`, `.js`, `.custom-ext`, ...) should be parsed as TS (or TSX, depeding on the `isTSX` option).